Node/core-DirectLineWebSockets/DirectLineClient/app.js

```diff
@@ -0,0 +1,195 @@
+var Swagger = require('swagger-client');
+var open = require('open');
+var rp = require('request-promise');
+
+// Config settings
+var useW3CWebSocket = false;
+var directLineSecret = 'DIRECTLINE_SECRET';
+var directLineClientName = 'DirectLineClient';
+var directLineSpecUrl = 'https://docs.botframework.com/en-us/restapi/directline3/swagger.json';
+
+process.argv.forEach(function (val, index, array) {
+  if (val === 'w3c') {
+    useW3CWebSocket = true;
+  }
+});
+
+var directLineClient = rp(directLineSpecUrl)
+    .then(function (spec) {
+        // Client
+        return new Swagger({
+            spec: JSON.parse(spec.trim()),
+            usePromise: true
+        });
+    })
+    .then(function (client) {
```

Comment:
Can you also illustrate using the Token API to exchange the client's secret for a token?